### PR TITLE
D8CORE-642: Stanford Card Link display option.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Adds a drop-down to allow the user to choose a display option for the link.

# Review By (Date)
- October 29

# Urgency
- Medium. Nice to have for the Pilot.

# Steps to Test

1. Check out this branch.
2. Do a full install `blt drupal:install`
3. Log in and edit the home page node
4. Edit the Card paragraph and validate that the link display drop-down is available
5. Choose the `Action Link` option and save the node
6. Validate that the card displays a link
7. Edit the card again and choose `Button`
8. Save the node again and validate that it is a button.
9. Run tests

# Affected Projects or Products
- Stanford Page
- Stanford Card Paragraph
- D8CORE
- Jackalope

# Associated Issues and/or People
- D8CORE-642

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_paragraph_card/stanford_paragraph_card.module [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  // have the card_button_label region in the display just quietly end here.
+  if (!isset($variables['content']['#ds_configuration']['regions']['card_button_label'])) {
+    return;
+  }
+
+  // The key to which fields the link should go in to.
+  // By default, the display will render a button so we will only have to change
+  // it when the value is set to action. This behavior should be on all view
+  // modes.
+  $button_display = $paragraph->get('su_card_link_display')->getValue();
```
Comment (pookmish) [inline modules/stanford_paragraph_card/stanford_paragraph_card.module]: you can use `->getString()` which will give you just the `value` of the field.

File: modules/stanford_paragraph_card/stanford_paragraph_card.module [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  // You can stop here if this is not a paragraph type.
+  if (!isset($variables['content']['#paragraph'])) {
+    return;
+  }
+
+  // Define the paragraph object variable for shorter code.
+  $paragraph = $variables['content']['#paragraph'];
+
+  // You can also stop if the paragraph is not a card.
+  if ($paragraph->getType() !== "stanford_card") {
```
Comment (pookmish) [inline modules/stanford_paragraph_card/stanford_paragraph_card.module]: shouldn't this be `->bundle()`

Comment (sherakama) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: https://git.drupalcode.org/project/paragraphs/blob/8.x-1.x/src/ParagraphInterface.php#L102 

_shrugs_

File: modules/stanford_paragraph_card/stanford_paragraph_card.module (hunk 1 above)
Comment (sherakama) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: handy dandy.

File: modules/stanford_paragraph_card/stanford_paragraph_card.module (hunk 2 above)
Comment (pookmish) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: That method just calls the core method.
https://git.drupalcode.org/project/paragraphs/blob/8.x-1.x/src/Entity/Paragraph.php#L304

Comment (pookmish) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: I think using `->bundle()` will be more universal and easier to grep if necessary.

Comment (sherakama) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: https://git.drupalcode.org/project/paragraphs/blob/8.x-1.x/src/Entity/Paragraph.php#L303

Not sure why they wrapped that.

Comment (pookmish) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: yeah kinda dumb IMO

Comment (sherakama) [reply modules/stanford_paragraph_card/stanford_paragraph_card.module]: > I think using ->bundle() will be more universal and easier to grep if necessary.

yeah, I can get on board with that.

**Comment by sherakama** [comment]
@pookmish 

Revised the `->bundle()` thing.
Added test.

**Comment by pookmish** [comment]
cool. i'll run through manual tests here shortly. after we merge this, we should make a deployment to production.

**Pull request merged**